neqo-http3/src/features/extended_connect/webtransport_session.rs

```diff
@@ -0,0 +1,513 @@
+            .set_new_listener(Box::new(stream_event_listener.clone()));
+        Self {
+            control_stream_recv,
+            control_stream_send,
+            stream_event_listener,
+            session_id,
+            state: SessionState::Active,
+            frame_reader: FrameReader::new(),
+            events,
+            send_streams: BTreeSet::new(),
+            recv_streams: BTreeSet::new(),
+            role,
+        }
+    }
+
+    /// # Errors
+    /// The above function can only fail if supplied headers are not valid http headers.
+    /// # Panics
+    /// `control_stream_send` implements the  http specific functions and `http_stream()`
+    /// will never return `None`.
+    pub fn send_request(&mut self, headers: &[Header], conn: &mut Connection) -> Res<()> {
+        self.control_stream_send
+            .http_stream()
+            .unwrap()
+            .send_headers(headers, conn)
+    }
+
+    fn receive(&mut self, conn: &mut Connection) -> Res<(ReceiveOutput, bool)> {
+        qtrace!([self], "receive control data");
+        let (out, _) = self.control_stream_recv.receive(conn)?;
+        debug_assert!(out == ReceiveOutput::NoOutput);
+        self.maybe_check_headers();
+        self.read_control_stream(conn)?;
+        Ok((ReceiveOutput::NoOutput, self.state == SessionState::Done))
+    }
+
+    fn reset(&mut self, close_type: CloseType) {
+        self.close(close_type);
+    }
+
+    fn header_unblocked(&mut self, conn: &mut Connection) -> Res<(ReceiveOutput, bool)> {
+        let (out, _) = self
+            .control_stream_recv
+            .http_stream()
+            .unwrap()
+            .header_unblocked(conn)?;
+        debug_assert!(out == ReceiveOutput::NoOutput);
+        self.maybe_check_headers();
+        self.read_control_stream(conn)?;
+        Ok((ReceiveOutput::NoOutput, self.state == SessionState::Done))
+    }
+
+    fn maybe_update_priority(&mut self, priority: Priority) -> bool {
+        self.control_stream_recv
+            .http_stream()
+            .unwrap()
+            .maybe_update_priority(priority)
+    }
+
+    fn priority_update_frame(&mut self) -> Option<HFrame> {
+        self.control_stream_recv
+            .http_stream()
+            .unwrap()
+            .priority_update_frame()
+    }
+
+    fn priority_update_sent(&mut self) {
+        self.control_stream_recv
+            .http_stream()
+            .unwrap()
+            .priority_update_sent();
+    }
+
+    fn send(&mut self, conn: &mut Connection) -> Res<()> {
+        self.control_stream_send.send(conn)?;
+        if self.control_stream_send.done() {
+            self.state = SessionState::Done
+        }
+        Ok(())
+    }
+
+    fn has_data_to_send(&self) -> bool {
+        self.control_stream_send.has_data_to_send()
+    }
+
+    fn done(&self) -> bool {
+        self.state == SessionState::Done
+    }
+
+    fn handle_stop_sending(&mut self, close_type: CloseType) {
+        self.close(close_type);
```

Comment:
This function does exactly the same as `reset`. Maybe we can remove one of them.

Reply:
Nice catch, both of them can be remove. They only call close().